### PR TITLE
docs(icons): Improve icon docs

### PR DESCRIPTION
**Pull request by tjgupta**
Added index pages for icons that were a cohesive set of icons. May attempt the `general` set next, though I'm not sure whether this type of documentation will work for that set... 

Note: somehow I got an updated translations file during the push. Maybe some other push did `--no-verify`?

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tjgupta has signed the CLA. Thanks for the pull request!

File: examples/src/IconsExample.js [hunk 1]
```diff
@@ -42,7 +45,7 @@ class IconsExample extends React.Component<Props, State> {
                         const Component = icon.component;
                         return (
                             <div className="icon" key={icon.name}>
-                                <Component />
+                                {icon.content ? icon.content() : <Component />}
```
Comment (tjgupta) [inline examples/src/IconsExample.js]: To allow for docs that need multiple variations of a single component displayed, like this:

Screen Shot 2019-03-11 at 9 23 19 PM

File: src/icons/annotations/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+**Annotation icons**
```
Comment (jstoffan) [inline src/icons/annotations/index.md]: Does Github display `index.md` files when you navigate to the folder via a browser? If not, could we name these centralized files `README.md`?

Comment (tjgupta) [reply src/icons/annotations/index.md]: Nope, doesn't seem like it. I switched them to `README.md`.

**Pull request merged**